pkg/services/datasources/models.go

```diff
@@ -150,60 +150,54 @@ func (e ErrDatasourceSecretsPluginUserFriendly) Error() string {
 // ----------------------
 // COMMANDS
 
-// Also acts as api DTO
-type AddDataSourceCommand struct {
+// The input values shared with both Add+Update commands (also part of the DTO)
+// This part of the command can share validation between add and update actions
+type BaseDataSourceCommand struct {
```

Comment:
the name may be a bit confusing because it could mean that `DeleteDatasource` also extends this one. Maybe something like `BaseWriteDataSourceCommand`?